pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -362,6 +388,190 @@ func TestReentrantRollback(t *testing.T) {
 	f.run(testutil.GetKey(d, t))
 }
 
+func recreateDeployment(deployments []*apps.Deployment, _ []*apps.ReplicaSet) {
+	d := deployments[0]
+	d.Spec.Strategy.Type = apps.RecreateDeploymentStrategyType
+	d.Spec.Strategy.RollingUpdate = nil
+}
+
+func rollingDeployment(maxUnavailable, maxSurge *intstr.IntOrString) func([]*apps.Deployment, []*apps.ReplicaSet) {
+	return func(deployments []*apps.Deployment, _ []*apps.ReplicaSet) {
+		d := deployments[0]
+		d.Spec.Strategy.Type = apps.RollingUpdateDeploymentStrategyType
+		d.Spec.Strategy.RollingUpdate = &apps.RollingUpdateDeployment{
+			MaxUnavailable: maxUnavailable,
+			MaxSurge:       maxSurge,
+		}
+	}
+}
+
+func TestUpdateConditions(t *testing.T) {
+	type objectsFunc func(t *testing.T) ([]*apps.Deployment, []*apps.ReplicaSet)
+	type wantActionsFunc func(t *testing.T, f *fixture, deployments []*apps.Deployment, replicaSets []*apps.ReplicaSet)
+	type modifierFunc func([]*apps.Deployment, []*apps.ReplicaSet)
+
+	simpleReplicaSetStatusFn := func(specReplicas, replicas, ready, available int32, fns ...modifierFunc) objectsFunc {
+		return func(t *testing.T) ([]*apps.Deployment, []*apps.ReplicaSet) {
+			d := newDeployment("foo", int(specReplicas), nil, nil, nil, map[string]string{"foo": "bar"})
```

Comment:
note: by default this is set to maxUnavailable = 25% and maxUnavailable = 25% in apiserver

in this case this will be then set to maxUnavailbale = 1 in `ResolveFenceposts`

If we are testing only this case, maybe we should say so explicitely in the test description/function name